src/openrct2/ride/RideRatings.cpp

```diff
@@ -158,13 +158,28 @@ static void ride_ratings_update_state(RideRatingUpdateState& state)
 static void ride_ratings_update_state_0(RideRatingUpdateState& state)
 {
     ride_id_t currentRide = state.CurrentRide;
-
-    currentRide++;
-    if (currentRide >= MAX_RIDES)
+    // It is possible that the existing ride being calculated has
+    // been removed or due to invalid import. For both, reset
+    // ratings check at the start
+    if (get_ride(currentRide) == nullptr)
```

Comment:
This could potentially cause problems if the user has a lot of rides, as the ones with low IDs will be updated more frequently if the counter resets. Would it be possible to just change this to go to the next available ID instead?

Reply:
In this hypothetical the user has to be deleting specific rides at specific timings during their update. That's not really realistic. You can't even spam create and delete as you would need to wait for ride measurements. I just can't see how this could ever be an issue. I can look see how the iterator handles the null but it seems overkill.

Reply:
You only lose one tick per unoccupied ride slot, although I guess that adds up. Might be good to move this to a function called, `GetNextRideToCalculate`